Fix broken player if mounted path had a trailing slash.

fungos opened the pull request:
The query on browse_songs wasn't returning any results as in the database they 
were save without the trailing slash.

Add two complementary output messages in case of errors trying to
create SharedData and SharedConfig where on Linux XDG defaults to
readonly system wide paths based on XDG_CONFIG_DIRS and XDG_DATA_DIRS
default values from spec (/etc/xdg and /usr/share for example).

And a minor change in the api to use forward slash.

agersant commented on src/api.rs:
```diff
@@ -82,7 +82,7 @@ pub fn get_api_handler(collection: Arc<Collection>) -> Mount {
 
 fn path_from_request(request: &Request) -> Result<PathBuf> {
-	let path_string = request.url.path().join("\\");
+	let path_string = request.url.path().join("/");
```
I think this would break on Windows instead because the DB stores path using backslashes on that platform. Can you change this to using path::MAIN_SEPARATOR instead?

agersant commented on src/config.rs:
```diff
@@ -193,9 +193,11 @@ impl Config {
 
 fn clean_path_string(path_string: &str) -> path::PathBuf {
 	let separator_regex = regex::Regex::new(r"\\|/").unwrap();
+	let trailing_regex = regex::Regex::new(r"(/)+$").unwrap();
```
Similar note here, we should also get rid of trailing backslashes. This change will work on Linux because the first `replace_all()` will convert everything to forward slashes but not on Windows.

fungos replied:
Sure, I've brainfarted on these. But at the same time I believe Rust interpret these correctly. In fact, I think you wouldn't need any platform differences at all if dealing only with Path/PathBuf everywhere.

agersant replied:
There's a few places I have to deal with raw strings:

- What ultimately goes in the DB
- Input that reaches the API
- User input from the config file

I decided to store paths in the DB using the native separator (whichever one I get from filesystem calls), which means the DB can deal with Path/PathBuf and do messy string handling. For the other two, I have no control over what comes in the other two so I think some sanitization is required. Those two code sites are precisely the ones you fixed so I think we should be in the clear from here, since after these gateways everything does use Path/Pathbuf.

Thanks again for the fixes!

fungos commented:
Yes, I got it to work. I believe the Path handling may be improved as per my other comment. But the thing that annoyed most was the use of the Shared places instead of the User ones, at least on default Linux installs users don't have right to write to it. Also I had some issues with Blankenberge ID3 tags on some files letting it hard to see what was what, but that is a more complex problem.

fungos commented:
Blank* ID3 tags. :)

fungos commented:
Fixed the issues, changed the code to remove traiiling spaces by using path iterator.
You may want to set appveyor in your project, here an example: https://github.com/starkat99/appveyor-rust/

agersant commented:
Blank tag handling is on the radar, but I think that's something I'll handle client side. I'd rather give clients accurate info about what's in the files and let them display whatever looks nice as a replacement (eg. filename, Unknown Artist, etc.).
